### PR TITLE
TASK-50543:New UI for settings page

### PR DESCRIPTION
**Pull request by GouadriaHanen**

### CODE REVIEW

File: webapp/portlet/src/main/webapp/WEB-INF/gatein-resources.xml [hunk 1]
```diff
@@ -724,6 +724,38 @@
     </module>
   </portlet>
 
+  <portlet>
+    <name>UserSettingMenu</name>
```
Comment (plamarque) [inline webapp/portlet/src/main/webapp/WEB-INF/gatein-resources.xml]:
```suggestion
    <name>UserSettingsMenu</name>
```

File: webapp/portlet/src/main/webapp/WEB-INF/gatein-resources.xml [hunk 2]
```diff
@@ -724,6 +724,38 @@
...
+    <module>
+      <script>
+        <minify>false</minify>
+        <path>js/userSettingMenu.bundle.js</path>
```
Comment (plamarque) [inline webapp/portlet/src/main/webapp/WEB-INF/gatein-resources.xml]:
```suggestion
        <path>js/userSettingsMenu.bundle.js</path>
```

File: webapp/portlet/src/main/webapp/WEB-INF/jsp/userSettingMenu.jsp [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<div class="VuetifyApp">
+  <div id="UserSettingMenu">
+    <script type="text/javascript">
+      require(['PORTLET/social-portlet/UserSettingMenu'], app => app.init());
```
Comment (plamarque) [inline webapp/portlet/src/main/webapp/WEB-INF/jsp/userSettingMenu.jsp]:
```suggestion
      require(['PORTLET/social-portlet/UserSettingsMenu'], app => app.init());
```

File: webapp/portlet/src/main/webapp/WEB-INF/portlet.xml [hunk 4]
```diff
@@ -293,6 +293,23 @@
     </portlet-info>
   </portlet>
 
+  <portlet>
+    <portlet-name>UserSettingMenu</portlet-name>
```
Comment (plamarque) [inline webapp/portlet/src/main/webapp/WEB-INF/portlet.xml]:
```suggestion
    <portlet-name>UserSettingsMenu</portlet-name>
```

File: webapp/portlet/src/main/webapp/WEB-INF/portlet.xml [hunk 5]
```diff
@@ -293,6 +293,23 @@
...
+    <portlet-class>org.exoplatform.commons.api.portlet.GenericDispatchedViewPortlet</portlet-class>
+    <init-param>
+      <name>portlet-view-dispatched-file-path</name>
+      <value>/WEB-INF/jsp/userSettingMenu.jsp</value>
```
Comment (plamarque) [inline webapp/portlet/src/main/webapp/WEB-INF/portlet.xml]:
```suggestion
      <value>/WEB-INF/jsp/userSettingsMenu.jsp</value>
```

File: webapp/portlet/src/main/webapp/vue-apps/user-setting-menu/components/UserSettingMenu.vue [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<template>
+  <v-app id="UserSettingMenu">
```
Comment (plamarque) [inline webapp/portlet/src/main/webapp/vue-apps/user-setting-menu/components/UserSettingMenu.vue]:
```suggestion
  <v-app id="UserSettingsMenu">
```

File: webapp/portlet/src/main/webapp/vue-apps/user-setting-menu/initComponents.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import UserSettingMenu from './components/UserSettingMenu.vue';
```
Comment (plamarque) [inline webapp/portlet/src/main/webapp/vue-apps/user-setting-menu/initComponents.js]:
```suggestion
import UserSettingsMenu from './components/UserSettingsMenu.vue';
```

File: webapp/portlet/src/main/webapp/vue-apps/user-setting-menu/initComponents.js [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+
+const components = {
+  'user-setting-menu': UserSettingMenu,
```
Comment (plamarque) [inline webapp/portlet/src/main/webapp/vue-apps/user-setting-menu/initComponents.js]:
```suggestion
  'user-settings-menu': UserSettingsMenu,
```

File: webapp/portlet/src/main/webapp/vue-apps/user-setting-menu/main.js [hunk 9]
```diff
@@ -0,0 +1,37 @@
+import './initComponents.js';
+
+
+// get overrided components if exists
+if (extensionRegistry) {
+  const components = extensionRegistry.loadComponents('UserSettingMenu');
```
Comment (plamarque) [inline webapp/portlet/src/main/webapp/vue-apps/user-setting-menu/main.js]:
```suggestion
  const components = extensionRegistry.loadComponents('UserSettingsMenu');
```

File: webapp/portlet/src/main/webapp/vue-apps/user-setting-menu/main.js [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+  }
+}
+
+
+const lang = eXo && eXo.env && eXo.env.portal && eXo.env.portal.language || 'en';
+const url = `${eXo.env.portal.context}/${eXo.env.portal.rest}/i18n/bundle/locale.portlet.social.UserSettings-${lang}.json` ;
+
+document.dispatchEvent(new CustomEvent('displayTopBarLoading'));
+
+const appId = 'UserSettingMenu';
```
Comment (plamarque) [inline webapp/portlet/src/main/webapp/vue-apps/user-setting-menu/main.js]:
```suggestion
const appId = 'UserSettingsMenu';
```

File: webapp/portlet/src/main/webapp/vue-apps/user-setting-menu/main.js [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+  exoi18n.loadLanguageAsync(lang, url)
+    .then(i18n => {
+      const appElement = document.createElement('div');
+      appElement.id = appId;
+      // init Vue app when locale ressources are ready
+      Vue.createApp({
+        mounted() {
+          document.dispatchEvent(new CustomEvent('hideTopBarLoading'));
+        },
+        template: '<user-setting-menu/>',
```
Comment (plamarque) [inline webapp/portlet/src/main/webapp/vue-apps/user-setting-menu/main.js]:
```suggestion
        template: '<user-settings-menu/>',
```

File: webapp/portlet/webpack.common.js [hunk 12]
```diff
@@ -31,6 +31,7 @@ let config = {
     userSettingNotifications: './src/main/webapp/vue-apps/user-setting-notifications/main.js',
     userSettingSecurity: './src/main/webapp/vue-apps/user-setting-security/main.js',
     userSettingTimezone: './src/main/webapp/vue-apps/user-setting-timezone/main.js',
+    userSettingMenu: './src/main/webapp/vue-apps/user-setting-menu/main.js',
```
Comment (plamarque) [inline webapp/portlet/webpack.common.js]:
```suggestion
    userSettingsMenu: './src/main/webapp/vue-apps/user-settings-menu/main.js',
```

**Review by plamarque — CHANGES_REQUESTED** [review]
please use plural everywhere for settings : UserSettingsMenu

**Comment by GouadriaHanen** [comment]
TASK closed

**Pull request closed without merging**